rocfft: split base outputs for hydra

Madouura opened the pull request:
###### Description of changes
Tracking: #197885
Last thing I can do before giving up on trying to get this to cache on hydra.
Technically I can have rocfft build twice, but then we'd have to do the same for composable_kernel, which I've learned is just barely under the output limit.
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Madouura commented:
It built hipfft and I'm not rebuilding torch again, so consider nixpkgs-review checked please.

SuperSandro2000 commented on pkgs/development/libraries/rocfft/default.nix:
```diff
@@ -19,13 +19,22 @@
 
     outputs = [
       "out"
+      "zero"
+      "one"
+      "two"
+      "three"
```
We should probably prefix those with ``lib-``.

Madouura replied:
They are that way intentionally to prevent variable contamination.

Madouura replied:
I apologize. Sleep deprivation made me a bit dumb and stubborn.
It's fixed now.
In case it's relevant, bash does not allow hyphens in variable names and I don't think a hyphen is worth using a ``placeholder``.

Madouura committed:
rocfft: fix propagatedBuildInputs typo

propagatedBuildInputs has been incorrectly spelled for a while now.

Madouura committed:
rocfft: split base outputs for hydra

Split the base derivation's outputs to hopefully avoid hydra failing the base due to the output limit and not continuing with the rest.
This and all hydra commits for rocfft and composable_kernel will be reverted if this does not work.

Madouura commented:
ofborg failing seems to be nixpkgs-wide at the moment and does not seem to be related to this PR.
Having ofborg redo the checks should fix this once the root problem is fixed.

Madouura commented:
Result of `nixpkgs-review pr 214606` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>hipfft</li>
    <li>python310Packages.torchWithRocm</li>
    <li>rocfft</li>
  </ul>
</details>

Flakebi reviewed (approved):
Do the intermediate `rf-zero`..`rf-three` derivations have an purpose left?

Anyway, LGTM, it’s worth a try.

Madouura commented:
I honestly don't know, not getting much help from those with more know-how on the matter so this was more or less the last spaghetti noodle I could throw at the wall and see if it stuck.

SuperSandro2000 commented:
@ofborg eval
